src/commands/publish/mod.rs

```diff
@@ -23,6 +23,36 @@ use crate::settings::project::{Project, ProjectType};
 
 pub fn publish(user: &GlobalUser, project: &Project, release: bool) -> Result<(), failure::Error> {
     info!("release = {}", release);
+
+    let missing_fields = match (project.account_id.is_empty(), project.name.is_empty()) {
+        (true, true) => Some("an account_id and name"),
+        (true, false) => Some("an account_id"),
+        (false, true) => Some("a name"),
+        (false, false) => None,
+    };
+
+    //perform extra checks for route and zone_id for a publish --release
+    let missing_fields_release = match (release, &project.zone_id, &project.route) {
+        (true, None, None) => Some("a zone_id and route"),
+        (true, None, Some(_)) => Some("a zone_id"),
+        (true, Some(_), None) => Some("a route"),
+        (_, _, _) => None,
+    };
+
+    if let Some(mf) = missing_fields {
+        failure::bail!(
+            "Check your wrangler.toml, it is missing {} which is required to publish!",
+            mf
+        )
+    }
+
+    if let Some(mf) = missing_fields_release {
+        failure::bail!(
+            "Check your wrangler.toml, it is missing {} which is required to publish a release!",
```

Comment:
This phrasing feels a little awkward to me, but I'm also not sure how best to re-phrase it...

Reply:
i'd probably say something like "Your wrangler.toml is missing `X` field. This is is required to publish to <domain/subdomain>."

You'll probably want to switch the message on what they are publishing to... (zone or zoneless). these have different req's

You might also add the caution that they may have been wanting to publish with or without `--release` if the things they have in their wrangler.toml suggest that